tests/BasicSearchTests.FunctionalTests.Core/StartupFunctionalTests.cs

```diff
@@ -0,0 +1,90 @@
+        {
+            // Arrange
+            _client = new HttpClient { BaseAddress = new Uri(EnvironmentSettings.SearchServiceBaseUrl) };
+        }
+
+        [Fact]
+        public async Task Ready()
+        {
+            // Act
+            var response = await _client.GetAsync("/");
+            var content = await response.Content.ReadAsStringAsync();
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal(content, "READY");
+        }
+
+        [Fact]
+        public async Task InvalidEndpoint()
+        {
+            // Act
+            var response = await _client.GetAsync("/invalid");
+            var content = await response.Content.ReadAsStringAsync();
+
+            // Assert
+            Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
+            Assert.Equal(content, "UNRECOGNIZED");
+        }
+
+        [Fact]
+        public async Task IndexIsFresh()
+        {
+            var httpClient = new HttpClient();
+            var indexResponse = await httpClient.GetAsync(EnvironmentSettings.IndexBaseUrl);
+            indexResponse.EnsureSuccessStatusCode();
+
+            var endpointContent = await indexResponse.Content.ReadAsStringAsync();
+            var endpointsList = JsonConvert.DeserializeObject<ServiceEndpointsList>(endpointContent);
+            var resources = endpointsList.Resources;
+
+            string registrationBaseUrl = null;
+            for (int i = 0; i < resources.Count; i++)
+            {
+                if (resources[i].AtType == RegistrationBaseUrl)
+                {
+                    registrationBaseUrl = resources[i].AtId;
+                    break;
+                }
+            }
+
+            Assert.False(registrationBaseUrl == null, "Failed to get registration base url. Please check that index url is correct.");
+
+            var cursorUri = new Uri(new Uri(registrationBaseUrl), "cursor.json");
+            var registrationResponse = await httpClient.GetAsync(cursorUri);
+            registrationResponse.EnsureSuccessStatusCode();
+
+            var registrationCursorContent = await registrationResponse.Content.ReadAsStringAsync();
+            var lastRegistrationCommitTime = JsonConvert.DeserializeObject<RegistrationCursor>(registrationCursorContent).Value;
+            Assert.NotNull(lastRegistrationCommitTime);
+
+            var response = await _client.GetAsync("/search/diag");
+            var content = await response.Content.ReadAsAsync<SearchDiagResult>();
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.NotNull(content);
+
+            var diffTimes = lastRegistrationCommitTime.Subtract(content.CommitUserData.CommitTimeStamp).TotalHours;
+            //Last CommitTimeStamp for search service shouldn't be far off from the last registration timestamp.
+            Assert.True(diffTimes >= 0 && diffTimes <= 1, "Search index is ahead of last registration timestamp");
```

Comment:
Does this fail if a job is down? I would be a little careful about introducing a functional test that relies on the jobs being functional.